libs/ledger-live-common/src/hw/firmwareUpdate-repair.ts

```diff
@@ -152,9 +153,9 @@ const repair = (
                         });
 
                         const mcu = ManagerAPI.findBestMCU(
-                          finalFirmware.mcu_versions
-                            .map(id => validMcusForDeviceInfo.find(mcu => mcu.id === id))
-                            .filter(Boolean),
+                          mcus.filter(({ id }: McuVersion) =>
+                            finalFirmware.mcu_versions.includes(id),
+                          ),
```

Comment:
It's a rewrite, old code is a bad implementation.

Reply:
If there were tests for this function, it will pass, if there were not, should add them. Btw, the old code is not TypeScript and tricky filter has nothing to do here.

Reply:
Can this be done in a separate PR please ? this PR should stick to pure refactoring I think, it's big enough as is.
Firmware update is hard to test so we should be very careful with changes there, or at least add unit tests where some are missing.